### PR TITLE
[opinfo] use dtypes instead of dtypesIfCPU

### PR DESCRIPTION
**Pull request by kshitij12345**
Reland #67619

Replace usage of dtypesIfCPU with dtypes in OpInfo class and also make it a mandatory argument.

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -524,10 +524,13 @@ def __init__(self,
                  error_inputs_func=None,  # function to generate inputs that will throw errors
 
                  # the following metadata relates to dtype support and is tested for correctness in test_ops.py
-                 dtypes=floating_types(),  # dtypes this function is expected to work with
+                 dtypes,  # dtypes this function works with on the CPU,
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:527]: Thanks for @pmeier for the suggestion of making it without default. We can now get away with `assert dtypes is not None`

**Review by mruberry — APPROVED** [review]
Stamped

**Comment by mruberry** [comment]
Internal tools say this is hitting a merge conflict, sorry -- would you just rebase this and ping me, @kshitij12345?

**Comment by kshitij12345** [comment]
@mruberry this should be ready once the CI passes. Thanks!

**Pull request closed without merging**